### PR TITLE
Th 67/68/bug fix

### PR DESCRIPTION
**Pull request by lanceljr**
## Context
Resolved bug tickets filed by Caleb. Key issues involved:

- Mentors/Mentees being able to submit blank spaces only when editing text fields in their profile
- Error messages not disappearing even after user has input the correct values

JIRA ticket: 
1. https://teamhalpert.atlassian.net/jira/software/projects/TH/boards/1/backlog?selectedIssue=TH-67
2. https://teamhalpert.atlassian.net/jira/software/projects/TH/boards/1/backlog?selectedIssue=TH-68

## Approach

- Added extra validation to check if user submits blank spaces in text fields
- Reset error message state whenever user clicks on the "Save" button 

## Checklist

- [ ] My branch is up-to-date with the `main` branch
- [ ] Unit and Integration test written
- [ ] Acceptance Criteria considered and implemented
- [ ] Automated tests are passing

## Risks
Low

### CODE REVIEW

File: src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx [hunk 1]
```diff
@@ -101,6 +101,7 @@ function ProfessionalExperience(props: Props) {
   }
 
   const handleSave = async () => {
+    setErrors(defaultErrors)
     const newErrors: Errors = deepCopy(errors)
```
Comment (khyechun) [inline src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx:105]: this should deepcopy default errors

Comment (lanceljr) [reply src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx:104]: Ok I've updated the code, can take a look again @khyechun

Comment (khyechun) [reply src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx:104]: oi deepCopy(defaultErrors) pls

File: src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx [hunk 2]
```diff
@@ -102,6 +102,7 @@ function ProfessionalExperience(props: Props) {
 
   const handleSave = async () => {
     const newErrors: Errors = deepCopy(errors)
+    setErrors(newErrors)
```
Comment (khyechun) [inline src/features/Onboarding/component/ProfessionalExperience/ProfessionalExperience.tsx:105]: This should be removed, and moved to above line 133

File: src/features/Onboarding/component/Interests/Interests.tsx [hunk 3]
```diff
@@ -75,6 +75,7 @@ function Interests(props: Props) {
     })
   }
   const handleSave = async () => {
+    setErrors(defaultErrors)
```
Comment (khyechun) [inline src/features/Onboarding/component/Interests/Interests.tsx:78]: This can be removed

**Comment by lanceljr** [comment]
@khyechun can review again

**Review by khyechun — APPROVED** [review]

**Pull request merged**